### DISCUSSION TITLE
Help with downloading IPFS command line for Windows

### DISCUSSION THREAD

**Discussion by ChristianG77** [smartcontractkit/full-blockchain-solidity-course-py#225] [Q&A] [answered]
Hi all,

I was looking for some help downloading IPFS command line for Windows.

Following the instructions for Windows on https://docs.ipfs.io/install/command-line/#official-distributions:

Instructions 1 to 3 look alright as I get the following output:

PS C:\Users\user> C:\Users\user\Dropbox\00_CODE\00_APPS\00_IPFS\go-ipfs_v0.10.0.\ipfs.exe --version
ipfs version 0.10.0

Instructions 4 and 5, I access, copy, paste, and save the below at the bottom of the profile.ps1 file.

> Path
> ----
> C:\Users\user\Dropbox\00_CODE\00_APPS\00_IPFS\go-ipfs_v0.10.0

Instruction 6, I close, reopen, as well as restarted my laptop, however receive the following output when checking if the version is installed.

PS C:\Users\user\Dropbox\00_CODE\NFT-DEMO> cd ~
PS C:\Users\user> ipfs --version
ipfs : The term 'ipfs' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and 
try again.      
At line:1 char:1
+ ipfs --version
+ ~~~~
    + CategoryInfo          : ObjectNotFound: (ipfs:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

Any help would be greatly appreciated.

Thanks.

**Comment by cromewar** [comment, marked as answer]
Sometimes when you are modifying powershell profiles you must use double `\\` for the paths, however there is a much easier way to solve this:
![1](https://user-images.githubusercontent.com/20199180/138115684-07e24b36-efd1-47c3-ad1d-28e2596ad98f.png)

As you see I also installed it using the commands, so in order to add it to your PATH, use `pwd` to know the exact location and then add it to your `PATH` manually on Windows advanced configuration like this:

![image](https://user-images.githubusercontent.com/20199180/138116110-57eb49fe-8085-4325-b90d-faedb953212c.png)

Now as you can see `ipfs` works just fine (after re start the terminal):

![image](https://user-images.githubusercontent.com/20199180/138116246-451ee8e7-859b-42f8-a035-3376f9dbf894.png)

If you don't know how to manually add environment variables to your path, here is a step by step guide:
https://www.architectryan.com/2018/08/31/how-to-change-environment-variables-on-windows-10/

**Reply by cromewar** [reply]
Finally I highly recommend to use wsl2 (windows subsystem for linux) for blockchain development in general, it would make your life easier ;)

**Reply by ChristianG77** [reply]
Great work @cromewar that worked perfectly!  

Thanks so much for your help, that was awesome, I was stuck on that for about a day haha!

**Reply by cromewar** [reply]
I'm glad it worked :)

**Reply by PatrickAlphaC** [reply]
boom!
